SharpestChain/SharpestChain/data/Block.cs

```diff
@@ -32,5 +32,47 @@ public static Block fromJson(string json)
         {
             return JsonConvert.DeserializeObject<Block>(json);
         }
+
+        public bool Equals(Block that)
```

Comment:
You could add the following:

```
public override bool Equals (object block) 
{
    return base.Equals(obj) || (block instanceof Block ? this.Equals((Block) block) : false);
}
```
In addition, you can then remove all the ReferenceEquals stuff from the second method.